installer/installer.go

```diff
@@ -0,0 +1,821 @@
+	}
+	return fmt.Sprintf("The built-in dashboard can be accessed at http://dashboard.%s with login token %s", s.Domain.Name, s.DashboardLoginToken), nil
+}
+
+func (s *Stack) promptUseExistingStack(savedStack *Stack) bool {
+	if s.StackID == "" || s.StackName == "" || savedStack.NumInstances != s.NumInstances || savedStack.InstanceType != s.InstanceType || savedStack.Region != s.Region {
+		return false
+	}
+
+	clusterAddCmd, err := savedStack.ClusterAddCmd()
+	if err != nil {
+		return false
+	}
+	dashboardMsg, err := savedStack.DashboardLoginMsg()
+	if err != nil {
+		return false
+	}
+
+	if err := s.fetchStack(); err != nil {
+		return false
+	}
+
+	if !s.YesNoPromt(fmt.Sprintf("It appears you already have a cluster of this configuration (stack %s), would you like to continue?", s.StackName)) {
+		s.HandleEvent(fmt.Sprintf("Install the flynn-cli (see https://cli.flynn.io for instructions) and paste the line below into a terminal window:\n\n%s\n\n%s", clusterAddCmd, dashboardMsg))
+		return true
+	}
+	return false
+}
+
+func (s *Stack) RunAWS() error {
+	s.setDefaults()
+	if err := s.validateInputs(); err != nil {
+		return err
+	}
+
+	s.EventChan = make(chan *Event)
+	s.ErrChan = make(chan error)
+	s.Done = make(chan struct{})
+	s.InstanceIPs = make([]string, 0, s.NumInstances)
+	s.ec2 = ec2.New(s.Creds, s.Region, nil)
+	s.cf = cloudformation.New(s.Creds, s.Region, nil)
+
+	savedStack := &Stack{}
+	savedStack.load()
+	s.StackID = savedStack.StackID
+	s.StackName = savedStack.StackName
+	s.SSHKeyName = savedStack.SSHKeyName
+
+	go func() {
+		defer close(s.Done)
+
+		if s.promptUseExistingStack(savedStack) {
+			return
+		}
+
+		steps := []func() error{
+			s.createKeyPair,
+			s.allocateDomain,
+			s.fetchImageID,
+			s.createStack,
+			s.fetchStackOutputs,
+			s.configureDNS,
+			s.bootstrap,
+		}
+
+		for _, step := range steps {
+			if err := step(); err != nil {
+				s.persist()
+				s.HandleError(err)
+				return
+			}
+			if err := s.persist(); err != nil {
+				s.HandleError(err)
+			}
+		}
+	}()
+	return nil
+}
+
+func (s *Stack) HandleEvent(description string) {
+	s.eventMutex.Lock()
+	defer s.eventMutex.Unlock()
+	s.EventChan <- &Event{description}
+}
+
+func (s *Stack) HandleError(err error) {
+	s.errMutex.Lock()
+	defer s.errMutex.Unlock()
+	s.ErrChan <- err
+}
+
+func (s *Stack) fetchImageID() error {
+	s.HandleEvent("Fetching image manifest")
+	if err := func() error {
+		latestVersion, err := fetchLatestVersion()
+		if err != nil {
+			return err
+		}
+		var imageID string
+		for _, i := range latestVersion.Images {
+			if i.Region == s.Region {
+				imageID = i.ID
+				break
+			}
+		}
+		if imageID == "" {
+			return errors.New(fmt.Sprintf("No image found for region %s", s.Region))
+		}
+		s.ImageID = imageID
+		return nil
+	}(); err != nil {
+		s.HandleEvent(err.Error())
+		if s.ImageID != "" {
+			s.HandleEvent("Falling back to saved Image ID")
+			return nil
+		} else {
+			return err
+		}
+	}
```

Comment:
Instead of an anonymous function for processing, you can do this:

```go
func (s *Stack) fetchImageID() (err error) {
  defer func() {
    if err == nil {
      return
    }
    // error handling here
  }()